### PR TITLE
Implement formatting for C# 9 patterns

### PR DESCRIPTION
**Pull request by sharwell**
Contributes to #42368

### CODE REVIEW

**Comment by alrz** [comment]
I believe the formatter should be responsible to align multiline patterns in a meaningful manner. 
See https://github.com/dotnet/roslyn/pull/43596#issuecomment-618403732 for the current state of the results.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/SpacingFormattingRule.cs [hunk 1]
```diff
@@ -314,6 +320,24 @@ public override AbstractFormattingRule WithOptions(AnalyzerConfigOptions options
                 }
             }
 
+            // For spacing after the 'not' pattern operator
+            if (previousToken.Parent.IsKind(SyntaxKindEx.NotPattern))
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/SpacingFormattingRule.cs:324]: hrmm... i wouldn't control this with SpacingAroundBinaryOperator.  For one, it's not a binary operator.  For another, it's just a word, and should likely just always be forced to single space (like we would with much of our other prefix-keyword constructs).

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/TokenBasedFormattingRule.cs [hunk 2]
```diff
@@ -509,6 +509,16 @@ private static SyntaxList<UsingDirectiveSyntax> GetUsings(SyntaxNode node)
...
 
+            // Relational unary prefix operators within patterns
+            // < *
+            // <= *
+            // > *
+            // >= *
+            if (previousToken.IsKind(SyntaxKind.LessThanToken, SyntaxKind.LessThanEqualsToken, SyntaxKind.GreaterThanToken, SyntaxKind.GreaterThanEqualsToken)
+                && previousToken.Parent.IsKind(SyntaxKindEx.RelationalPattern))
+            {
+            }
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/TokenBasedFormattingRule.cs:520]: ?

Comment (sharwell) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/TokenBasedFormattingRule.cs:520]: Guess this didn't need anything 😆

Comment (sharwell) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/TokenBasedFormattingRule.cs:520]: Removed

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/SpacingFormattingRule.cs (hunk 1 above)
Comment (sharwell) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/SpacingFormattingRule.cs:324]: Updated

**Pull request merged**